com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/UniformBufferCollection.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace UnityEditor.ShaderFoundry
+{
+    internal class UniformBufferCollection
+    {
+        internal class UniformBufferObject
+        {
+            readonly public UniformDataSource DataSource;
+            readonly public string BufferName;
+
+            public ShaderBuilder Builder = new ShaderBuilder();
+
+            public UniformBufferObject(UniformDataSource dataSource, string bufferName)
+            {
+                DataSource = dataSource;
+                BufferName = bufferName;
+            }
+
+            public void AddDeclaration(ShaderBuilder outBuilder)
+            {
+                if (DataSource != UniformDataSource.Global)
```

Comment:
I think we'll use this path for `UniformDataSource.None` as well - is it worth a check here?